scio-core/src/main/java/com/spotify/scio/util/RemoteFileUtil.java

```diff
@@ -103,10 +107,37 @@ public Path download(URI src) {
    * @return {@link Path}s to the downloaded local files.
    */
   public List<Path> download(List<URI> srcs) {
+    return download(srcs, CONCURRENCY_LEVEL);
+  }
+
+  /**
+   * Download a batch of remote {@link URI}s in parallel, using at most numThreads to do so.
+   * `numThreads` may not be larger than the number of available processors * 4.
+   *
+   * @return {@link Path}s to the downloaded local files.
+   */
+  public List<Path> download(List<URI> srcs, int numThreads) {
+    final ExecutorService executor = Executors.newFixedThreadPool(numThreads);
```

Comment:
We should probably move this as transient member, initializing it in the constructor.
The class can implement `AutoClosable` to shutdown the thread pool.

This is mainly relevant for the `FileDownloadDoFn` where we don't want to create a new thread pool for every url batch.